packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -469,12 +471,16 @@ public void run(Future<String> tokenFuture) {
               } catch (ExecutionException e) {
                 if (e.getCause() instanceof UserRecoverableAuthException) {
                   if (shouldRecoverAuth && pendingOperation == null) {
-                    checkAndSetPendingOperation(METHOD_GET_TOKENS, result, email);
-                    Intent recoveryIntent =
-                        ((UserRecoverableAuthException) e.getCause()).getIntent();
-                    registrar
-                        .activity()
-                        .startActivityForResult(recoveryIntent, REQUEST_CODE_RECOVER_AUTH);
+                    Activity activity = registrar.activity();
+                    if (activity == null) {
+                      Log.w(TAG, "Cannot recover auth because app is not in foreground");
```

Comment:
Not sure about this kind of logging. It is not a widely used pattern for plugins.

Consider appending this string to the error you return to Dart.

Reply:
I've prepended it to the `Result`'s error message.